GrandCentralBoard/Widgets/Watch/WatchWidgetViewModel.swift

```diff
@@ -8,59 +8,119 @@ import UIKit
-    let selectedImage: UIImage?
-    let calendarName: String?
+    let centeredTimeText: String?
+    let alignedTimeText: String?
+    let eventText: NSAttributedString?
 
-    init(date: NSDate, timeZone: NSTimeZone, event: Event?, calendarName: String?) {
 
+    init(date: NSDate, timeZone: NSTimeZone, event: Event? = nil, calendarName: String? = nil) {
         if let event = event {
-            let eventComponents = WatchWidgetViewModel.componentsFromDate(event.time, timeZone: timeZone)
-            let selected = (eventComponents.minute / 5) + 1
-            selectedImage = UIImage(named: "s\(selected)")
-            let minutes = Int(event.time.timeIntervalSinceDate(NSDate()) / 60)
-            let isNow = minutes <= 1
-            meetingName = event.name
-            startsIn = isNow ? "is" : "starts in"
-            meetingETA = isNow ? "now" : "\(minutes) minutes"
+            let timeToEvent = event.time.timeIntervalSinceDate(date)
+            let minutesToEvent = Int(ceil(timeToEvent / 60)) // 10 seconds is "1 minute left"; 1 minute and 10 seconds is "2 minutes left"
+
+            centeredTimeText = nil
+            alignedTimeText = WatchWidgetViewModel.dateFormatterWithTimeZone(timeZone).stringFromDate(date)
+            eventText = WatchWidgetViewModel.descriptionForEvent(event, fromCalendar: calendarName, startingInMinutes: minutesToEvent)
+            watchFaceImage = WatchWidgetViewModel.watchFaceImageForEventStartingInMinutes(minutesToEvent)
         } else {
-            selectedImage = nil
-            meetingName = nil
-            meetingETA = nil
-            startsIn = nil
+            centeredTimeText = WatchWidgetViewModel.dateFormatterWithTimeZone(timeZone).stringFromDate(date)
+            alignedTimeText = nil
+            eventText = nil
+            watchFaceImage = nil
         }
+    }
+
+    private static func watchFaceImageForEventStartingInMinutes(minutesToEvent: Int) -> UIImage? {
+        guard minutesToEvent > 0 else { return nil }
+
+        let minutesInSlot = 5
+        let numberOfSlots = 12
+        let slotIndex = min(Int(ceil(Float(minutesToEvent) / Float(minutesInSlot))), numberOfSlots)
+
+        return UIImage(named: "\(slotIndex)")
+    }
+
+
+    // MARK: Event description formatting
 
-        let currentTimeComponents = WatchWidgetViewModel.componentsFromDate(date, timeZone: timeZone)
+    private static let nextMeetingInFont = UIFont.systemFontOfSize(24, weight: UIFontWeightSemibold)
+    private static let minutesFont = UIFont.systemFontOfSize(50, weight: UIFontWeightSemibold)
+    private static let nowFont = UIFont.systemFontOfSize(50, weight: UIFontWeightSemibold)
+    private static let calendarNameFont = UIFont.systemFontOfSize(22, weight: UIFontWeightBold)
+    private static let eventNameFont = UIFont.systemFontOfSize(22, weight: UIFontWeightBold)
+    private static let calendarNameTextColor = UIColor(red: 35/255, green: 208/255, blue: 165/255, alpha: 1)
```

Comment:
This color will be specified in an extension in #131 , it should be merged soon.